mpqp/execution/runner.py

```diff
@@ -178,6 +178,8 @@ def _run_single(
          Error: None
 
     """
+    circuit = circuit.hard_copy()
```

Comment:
If we put a hard_copy of the circuit here, we will need to go into each provider file and remove it each time we call `hard_copy` on the circuit (to avoid a double call). This may result in a task in Trello since some are done in other branches.

Reply:
I tested this, and apparently, the circuit is not updated correctly
```python
>>> from mpqp.all import *
>>> c = QCircuit([H(0), CNOT(1,3), BasisMeasure()])
>>> c.nb_cbits

>>> c.hard_copy().nb_cbits
4
>>> print(run(c,ATOSDevice.MYQLM_PYLINALG)) # we run the circuit, to call _run_single, which should update the circuit
Result: None, ATOSDevice, MYQLM_PYLINALG
 Counts: [481, 0, 0, 0, 0, 0, 0, 0, 543, 0, 0, 0, 0, 0, 0, 0]
 Probabilities: [0.4697266, 0, 0, 0, 0, 0, 0, 0, 0.5302734, 0, 0, 0, 0, 0, 0, 0]
 Samples:
  State: 0000, Index: 0, Count: 481, Probability: 0.4697265625
  State: 1000, Index: 8, Count: 543, Probability: 0.5302734375
 Error: 0.015603954651445045

>>> c.nb_cbits

>>> c
QCircuit([H(0), CNOT(1,3), BasisMeasure(, shots=1024)], nb_qubits=4, nb_cbits=None, label="None")
```

We see that after execution, the `nb_cbits` of the circuit is not updated, nor the BasisMeasure,

Reply:
It works well, it is not designed to update the circuits after a run. This applies to cases where we run the circuit and then change the number of qubits. However, I came up with a new idea for the instruction. Instead of providing an empty target list, we can directly specify the targets and introduce a new variable in the instruction, such as 'dynamic: bool = False'. When we update the number of qubits, we can check 'dynamic = True' and change the targets. So we don't have to use hard_copy . What do you think?

Reply:
I agree with idea, as discussed. It is fully implemented now, right?

Reply:
yes ready to review